### PR TITLE
T2D-311 — Test run for importing reference sequence, sample, taxonomy

### PR DESCRIPTION
**Pull request by tskir**
### [Diff link](https://github.com/tskir/metadata-ws/compare/t2d-291-store-sample...t2d-311-test-run?expand=1)

* Rewrite EGA to use accession-based reference sequence import, remove fixed lookup table
* Split by patch for GRC human and mouse assemblies
* Update tests for new patch splitting logic
* Force reference sequence import via API for all cases
* Migrate API retriever bean into the common configuration

### CODE REVIEW

**Comment by tskir** [comment]
For details about patch splitting, see discussion in the ticket https://www.ebi.ac.uk/panda/jira/browse/T2D-311

File: metadata-load/src/main/java/uk/ac/ebi/ampt2d/metadata/importer/converter/ReferenceSequenceConverter.java [hunk 1]
```diff
@@ -27,9 +27,18 @@
 
     @Override
     public ReferenceSequence convert(AssemblyType assemblyType) {
+        String refName = assemblyType.getNAME();
+        String patch = null;
+        if (refName != null) {  // Attempt to extract patch from refName (split by dot)
```
[the remark this answers is not in the document]
Comment (tskir) [reply metadata-load/src/main/java/uk/ac/ebi/ampt2d/metadata/importer/converter/ReferenceSequenceConverter.java]: Fixed in https://github.com/EBIvariation/metadata-ws/pull/58/commits/8f64b9208eac3adcd37a3214081340d07a2f74b0

**Comment by tskir** [comment]
Changes:
* Rewrite EGA to use accession-based reference sequence import, remove fixed lookup table
* Split by patch for GRC human and mouse assemblies
* Update tests for new patch splitting logic
* Force reference sequence import via API for all cases
* Migrate API retriever bean into the common configuration

Ready for the next iteration of review.

File: metadata-load/src/main/java/uk/ac/ebi/ampt2d/metadata/importer/database/SraObjectsImporterThroughDatabase.java [hunk 2]
```diff
@@ -200,21 +141,6 @@ protected Study extractAnalysisFromStudy(StudyType studyType, Study study) {
      *
      * @return combined accession and refname. Example: "GCA_000001405.1#GRCh37"
      */
-    @Override
```
Comment (selvaebi) [inline metadata-load/src/main/java/uk/ac/ebi/ampt2d/metadata/importer/database/SraObjectsImporterThroughDatabase.java]: Comments haven't been removed.

File: metadata-load/src/test/java/uk/ac/ebi/ampt2d/metadata/importer/database/SraObjectsImporterThroughDBTest.java [hunk 3]
```diff
@@ -86,14 +86,15 @@ public void importStudy() throws Exception {
     @Test
     @Category(OracleDbCategory.class)
     public void importAnalysisObject() throws Exception {
-        Analysis analysis = sraObjectImporter.importAnalysis("ERZ000011");
+        Analysis analysis = sraObjectImporter.importAnalysis("ERZ019262");
```
Comment (selvaebi) [inline metadata-load/src/test/java/uk/ac/ebi/ampt2d/metadata/importer/database/SraObjectsImporterThroughDBTest.java]: Can you please add test case for analysis with empty refSequence accession.

File: metadata-load/src/main/java/uk/ac/ebi/ampt2d/metadata/importer/database/SraObjectsImporterThroughDatabase.java (hunk 2 above)
Comment (tskir) [reply metadata-load/src/main/java/uk/ac/ebi/ampt2d/metadata/importer/database/SraObjectsImporterThroughDatabase.java]: Whoops, fixed in https://github.com/EBIvariation/metadata-ws/pull/58/commits/24cc13de55fe41b8224bf3eb6eb0facf6f6e3d92

File: metadata-load/src/test/java/uk/ac/ebi/ampt2d/metadata/importer/database/SraObjectsImporterThroughDBTest.java (hunk 3 above)
Comment (tskir) [reply metadata-load/src/test/java/uk/ac/ebi/ampt2d/metadata/importer/database/SraObjectsImporterThroughDBTest.java]: Added in https://github.com/EBIvariation/metadata-ws/pull/58/commits/5e089fff7bc217fd3c7eb42c4d4e3c0c805aece9

**Comment by tskir** [comment]
Fixed both comments, ready for the next iteration of review.
